--- freeswitch.orig/src/mod/endpoints/mod_sofia/mod_sofia.h	2015-11-17 16:40:40.000000000 +0300
+++ freeswitch/src/mod/endpoints/mod_sofia/mod_sofia.h	2017-08-04 11:23:37.252575484 +0300
@@ -335,6 +335,7 @@
 	TFLAG_REINVITED,
 	TFLAG_PASS_ACK,
 	TFLAG_KEEPALIVE,
+	TFLAG_MODIFY_RTP,
 	/* No new flags below this line */
 	TFLAG_MAX
 } TFLAGS;
--- freeswitch.orig/src/mod/endpoints/mod_sofia/sofia.c	2015-11-17 16:40:40.000000000 +0300
+++ freeswitch/src/mod/endpoints/mod_sofia/sofia.c	2017-08-03 19:47:35.307737555 +0300
@@ -1510,6 +1510,7 @@
 		sofia_handle_sip_i_info(nua, profile, nh, session, sip, de, tags);
 		break;
 	case nua_i_update:
+		sofia_set_flag(tech_pvt, TFLAG_MODIFY_RTP);
 		break;
 	case nua_r_update:
 		if (session && tech_pvt && locked) {
@@ -6672,6 +6673,10 @@
 				//	switch_core_media_set_sdp_codec_string(session, r_sdp, status < 200 ? SDP_TYPE_REQUEST : SDP_TYPE_RESPONSE);
 				//}
 
+				if (sofia_test_flag(tech_pvt, TFLAG_MODIFY_RTP)) {
+					switch_core_media_proxy_remote_addr(session, r_sdp);
+					sofia_clear_flag(tech_pvt, TFLAG_MODIFY_RTP);
+				}
 				sofia_glue_pass_sdp(tech_pvt, (char *) r_sdp);
 				sofia_set_flag(tech_pvt, TFLAG_NEW_SDP);
 
